### DISCUSSION TITLE
Use mouse middle click to close tab

### DISCUSSION THREAD

**Discussion by pl4wdroid** [sumatrapdfreader/sumatrapdf#2854] [Ideas]
Hi there, and great work with Sumatra PDF, been a user since ... eh, can't remember :)

Would it be possible to get a mouse middle click (mouse wheel click) to close the current tab (under the mouse cursor), like with Internet browser tabs?

I use this every day to close browser tabs but not only as most apps using tabs have now integrated that capability, even Windows to close apps from the taskbar.

Would love to have it in Sumatra as I'm often left frustrated when clicking and the tab's not closing. You then have to move cursor to the tab's red cross or use CTRL+W.

Cheers!

**Comment by GitHubRulesOK** [comment]
should still be working I could even close the home tab with middle click. It is only on release so any move away from down click will cancel it

**Reply by pl4wdroid** [reply]
Hi, and thanks. Indeed the first mouse middle click on a tab will activate the red cross, second click will close the tab.

Has it changed? I remember it required a single mouse middle click to close a tab (not sure as I do it without even thinking hundreds of times a day if I consider all the apps I'm using that are using tabs)

**Reply by GitHubRulesOK** [reply]
AFAIK it has been "lock and load" or "prime and fire" e.g. its the middle button down **and** up that triggers closure hence early complaints when user gets new kit and activity zone is tighter than expected on hi-res screens.

**Reply by pl4wdroid** [reply]
Hi, and thanks again.

> e.g. its the middle button down and up that triggers closure

You mean pressing the middle button (down), then fully releasing it (up), right?

In my case and tested on a few computers, it requires that once (down, then up) to activate the red cross at the right end of a tab (cross goes from whatever light grey color to red) then a second time (down, then up) to actually close it.
Moving the mouse cursor out of the tab after the first "full click" will reset the cross to 'whatever light grey color' and you have to start again.

I could swear it wasn't like this before, as in: a single middle click on the tab (down, then up) would close it, just like any Internet browser tab if you see what I mean. Am I losing my mind? /scared face/

**Reply by GitHubRulesOK** [reply]
Ahh 
confirmed recent pre-release tab behaviors are odd, 
there are a few more related issues since around since pre-rel 3.5.15226 (about 15th July)

@kjk unsure if related to other tab problems with home tab or dpi quirks but this  mid button feature was not showing red closing in 15224 before needing two click sequence in 15226

**Comment by pl4wdroid** [comment]
Nothing new about this one? So frustrating :(
Cheers!

**Reply by GitHubRulesOK** [reply]
Hmm it works for me one middle click but now there is no red X just a white one since when tabs were last changed.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/a8ee1f2c-6106-44f3-b5dc-18b8f0ba0fd4)

**Comment by pl4wdroid** [comment]
v3.4.6 64-bit here, still requires 2 middle mouse clicks on all my
computers -.-
First click makes the tab cross red, second click closes the tab.

Cheers!

**Reply by GitHubRulesOK** [reply]
all the changes can only be in[ pre-release](https://www.[sumatrapdfreader.org/prerelease](https://www.sumatrapdfreader.org/prerelease)) and draggable coloured tabs both NOT listed here !! changed many other functions see https://www.sumatrapdfreader.org/docs/Version-history

so not closed here yet as those are very experimental but is available in pre-release for 3.5

**Comment by kjk** [comment]
Fixed in https://github.com/sumatrapdfreader/sumatrapdf/commit/d973b3eab289b353b0b9caad7452bba00063ad8e

**Reply by pl4wdroid** [reply]
Awesome, great job and thanks!

**Reply by Industry** [reply]
Nicely done!
